### PR TITLE
Display resolved property value with placeholder

### LINKED ISSUE #2760: Display full property names with resolved default value
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
Instead of just resolving the property in the configuration, I believe it would benefit developers if they can see the property name as well as the resolved value. There is a standard which Tomcat/Spring implement to achieve this `${property.name:-default value}`.
The framework should be able to parse properties as stated above, so with a copy paste nothing will break.

Mainly for the Ladybug and potentially the Frank!Flow.

**Reporter**
Niels

### PR DESCRIPTION
**Pull request by alisihab**
Closes #2760

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/util/StringResolver.java [hunk 1]
```diff
@@ -90,38 +92,47 @@ public static String getSystemProperty(String key, String def) {
...
-	public static String substVars(String val, Map props1, Map props2, List<String> propsToHide, String delimStart, String delimStop) throws IllegalArgumentException {
 
-		StringBuffer sbuf = new StringBuffer();
+	public static String substVars(String val, Map props1, Map props2, List<String> propsToHide, String delimStart, String delimStop) throws IllegalArgumentException {
+		return substVars(val, props1, props2, propsToHide, delimStart, delimStop, false);
+	}
 
-		int i = 0;
-		int j, k;
+	public static String substVars(String val, Map props1, Map props2, List<String> propsToHide, String delimStart, String delimStop, boolean displayWithValue) throws IllegalArgumentException {
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/util/StringResolver.java:102]: displayWithValue should be: resolveWithPropertyName

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/test/java/nl/nn/adapterframework/util/StringResolverTest.java [hunk 2]
```diff
@@ -72,36 +101,54 @@ public void resolveUsername() {
...
 	}
 
 	@Test
 	public void resolvePasswordNotAllowed() {
 		String result = StringResolver.substVars("${credential:password:alias3}", properties);
 		assertEquals("!!not allowed to expand credential of authAlias [alias3]!!", result);
+
+		result = StringResolver.substVars("${credential:password:alias3}", properties, true);
+		assertEquals("${credential:password:alias3:-!!not allowed to expand credential of authAlias [alias3]!!}", result);
 	}
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/StringResolverTest.java:153]: Can you also add a test when it cannot resolve the property? Aka there wont be a default value

File: core/src/test/resources/StringResolver.properties [hunk 3]
```diff
@@ -5,4 +5,7 @@ key4=${key1}.${key3}
 key5=${key5}
 key1_value1=value101
 testMultiResolve=one,two,three_${key1}${key2},my_${key3},StageSpecifics_${key4}
-cyclic=prefix ${cyclic} suffix
\ No newline at end of file
+cyclic=prefix ${cyclic} suffix
+log.dir=c:/temp
+testdata=${log.dir}/testdata
+mfh=${testdata}/mfh
```
Comment (nielsm5) [inline core/src/test/resources/StringResolver.properties:11]: Can you also add a property with a default value, so we can test that the framework can parse them without the `resolveWithPropertyName` flag enabled.

**Review by nielsm5 — CHANGES_REQUESTED** [review]

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/util/StringResolver.java [hunk 4]
```diff
@@ -176,14 +199,48 @@ public static String substVars(String val, Map props1, Map props2, List<String>
...
+			}
+			head = tail + delimStop.length();
+		}
+	}
+
+	/**
+	 * Resolves just the values of the properties in case a property key depends on other keys
+	 * e.g System.getProperty("prefix_${key:-value}") will find no matching data, this method extracts the 'value' for property lookup prefix_value
+	 */
+	private static String extractKeyValue(String key, String delimStart, String delimStop, String defualtValueSeparator) {
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/util/StringResolver.java:226]: oops I just noticed a typo: default

File: core/src/main/java/nl/nn/adapterframework/xml/AttributePropertyResolver.java [hunk 5]
```diff
@@ -44,6 +44,6 @@ public AttributePropertyResolver(ContentHandler handler, Properties properties,
 
 	@Override
 	public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
-		super.startElement(uri, localName, qName, new AttributesWrapper(attributes, v->StringResolver.substVars(v, properties, null, propsToHide)));
+		super.startElement(uri, localName, qName, new AttributesWrapper(attributes, v->StringResolver.substVars(v, properties, null, propsToHide, true)));
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/xml/AttributePropertyResolver.java:47]: Could you add a property to enable/disable this, with the default being off?

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/xml/AttributePropertyResolver.java [hunk 6]
```diff
@@ -22,12 +22,14 @@
...
 import org.xml.sax.SAXException;
 
+import nl.nn.adapterframework.util.AppConstants;
 import nl.nn.adapterframework.util.StringResolver;
 
 
 public class AttributePropertyResolver extends FullXmlFilter {
 	private Properties properties;
 	private List<String> propsToHide;
+	private static final boolean resolveWithPropertyName = AppConstants.getInstance().getBoolean("properties.display.name", false);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/xml/AttributePropertyResolver.java:32]:
```suggestion
	private static final boolean resolveWithPropertyName = AppConstants.getInstance().getBoolean("properties.resolve.withName", false);
```

**Review by nielsm5 — APPROVED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**